### ISSUE TITLE
Remove use of astropy-helpers submodule

### ISSUE THREAD

**Issue by mperrin** [spacetelescope/poppy#328]
See astropy/astropy-APEs#52 ("A roadmap for package infrastructure without astropy-helpers") which describes how the functionality in astropy-helpers can be achieved via more standardized tools in the Python ecosystem. 

This is clearly good and we should do it. 

See for instance https://github.com/spacetelescope/mosviz/pull/179 and https://github.com/spacetelescope/cubeviz/pull/492 for examples of PRs that remove astropy-helpers. We should do similarly here.

**Comment by joseph-long** [comment]
First of all, 🙌 

Second of all, I recently got an email from a would-be user that had copied the `poppy` dir into their Python path without going through setuptools, which skipped `version.py` generation, which made them think POPPY was broken. Just another data point against clever things in `setup.py` 😉

**Comment by mperrin** [comment]
I think I just wrote back to the same user - I had an email describing that problem with version.py waiting in my inbox when I got back from vacation. Were you able to get them up and running already?

**Comment by joseph-long** [comment]
Yes, I directed him to use the issue tracker in the future :)

**Comment by joseph-long** [comment]
(And he reports that everything worked after following the installation instructions to the letter)

**Comment by pllim** [comment]
You might want to do this sooner than later (or update your existing `astropy-helpers` version). @bhilbert4 reported this error trying to install `poppy` as part of `mirage` CI:

```
Searching for poppy>=0.9.0
Reading https://pypi.org/simple/poppy/
Downloading https://files.pythonhosted.org/packages/5e/dc/6799b3048282b37bb86fcc6ed1aa22d7c80d3264e92441153d02001ccb5e/poppy-0.9.0.tar.gz#sha256=43ccfb013690311022fe47626c503f2f7dbdb5aabe9f5df15ee828510088aa92
Best match: poppy 0.9.0
Processing poppy-0.9.0.tar.gz
Writing /tmp/easy_install-egmpb_3y/poppy-0.9.0/setup.cfg
Running poppy-0.9.0/setup.py -q bdist_egg --dist-dir /tmp/easy_install-egmpb_3y/poppy-0.9.0/egg-dist-tmp-389jg3ze
Sorry: TabError: inconsistent use of tabs and spaces in indentation (get_psf.py, line 19)
/tmp/easy_install-egmpb_3y/poppy-0.9.0/.eggs/astropy_helpers-2.0.11-py3.7.egg/astropy_helpers/setup_helpers.py:102: AstropyDeprecationWarning: Direct use of the adjust_compiler function in setup.py is deprecated and can be removed from your setup.py.  This functionality is now incorporated directly into the build_ext command.
  'command.', AstropyDeprecationWarning)
error: SandboxViolation: open('/home/docs/.astropy/config/astropy.4.0.cfg', 'wt') {'encoding': 'latin-1'}
The package setup script has attempted to modify files on your system
that are not within the EasyInstall build area, and has been aborted.
This package cannot be safely installed by EasyInstall, and may not
support alternate installation locations even if you run its setup
script by hand.  Please inform the package's author and the EasyInstall
maintainers to find out if a fix or workaround is available.
```

If you have already fixed this on dev, then you might need to do a new release. FYI.

**Comment by mperrin** [comment]
@shanosborne - do you have any availability to work on the de-astropy-helpering poppy and webbpsf? I believe this is now becoming more urgent with the latest astropy release. See above comment by @pllim.

**Comment by pllim** [comment]
@mperrin , just to be clear, `astropy` itself has not removed `astropy-helpers` yet (slated for 4.1). For your situation, simply updating `astropy-helpers` version to its latest tag might fix the issue. Although, if you are putting in the work already, might as well go all the way and apply APE 17? Tom R wrote a draft of a guide [here](https://demo.codimd.org/6TnoSu6PS1m-H8socv7JDg) in case you find it helpful.

**Comment by bhilbert4** [comment]
To be clear, I'm only seeing the error above during the Readthedocs build for Mirage. When installing Mirage, everything seems to build ok.

**Comment by pllim** [comment]
@bhilbert4 , can you please link the log to the failed doc build?

**Comment by bhilbert4** [comment]
https://readthedocs.org/projects/mirage-data-simulator/builds/10317921/

**Comment by shanosborne** [comment]
@mperrin yes I can get back to this issue!

**Comment by bhilbert4** [comment]
Mirage RTD builds are now passing. I removed a bunch of dependencies, including webbpsf/poppy  from the build, and everything looks good.

**Comment by mperrin** [comment]
Implemented in #349
